pcap4j-core/src/main/java/org/pcap4j/core/Pcaps.java

```diff
@@ -218,6 +223,33 @@ public static PcapHandle openOffline(
 
   /**
    *
+   * @param filePath "-" means stdin
+   * @param precision PCAP_TSTAMP_PRECISION_NANO or PCAP_TSTAMP_PRECISION_MICRO
+   * @return a new PcapHandle object.
+   * @throws PcapNativeException
+   */
+  public static PcapHandle openOfflineWithTstampPrecision(
+    String filePath, int precision
+  ) throws PcapNativeException {
+    if (filePath == null) {
+      StringBuilder sb = new StringBuilder();
+      sb.append("filePath: ").append(filePath);
+      throw new NullPointerException(sb.toString());
+    }
+
+    PcapErrbuf errbuf = new PcapErrbuf();
+    Pointer handle
+      = NativeMappings.pcap_open_offline_with_tstamp_precision(filePath, precision, errbuf);
```

Comment:
Here should be surrounded by try-catch like pcap_set_rfmon (after interface-mapping pcap_open_offline_with_tstamp_precision).